flow/src/main/java/org/jboss/hal/flow/Step.java

```diff
@@ -15,12 +15,22 @@
  */
 package org.jboss.hal.flow;
 
+import rx.Single;
+import rx.functions.Func1;
+
 /** Encapsulates one operation inside a flow */
-public interface Step<C> {
+public interface Step<C> extends Func1<C, Single<C>> {
 
     /**
      * Execute the step. Please make sure that you <strong>always</strong> call either {@link Control#proceed()} or
      * {@link Control#abort(String)}.
      */
     void execute(C context, Control control);
+
+    @Override default Single<C> call(C ctx) {
```

Comment:
this method is just to see that how to progressively refactor step so it doesn't need to extend anything, just any method that returns an RX type. I think that a massive refactor doesn't worth. Need to try it out in some task to see if it improve reliability and avoid to need to explain in the doc the expected input-output (bc it will be forced by the type system).